src/components/ModeSelector.tsx

```diff
@@ -45,10 +49,15 @@ export const ModeSelector = ({
   onSelectMode,
   selectedMode,
 }: ModeSelectorProps) => {
+  const theme = useTheme();
   const aggregate = aggregateResult(modeResults.map(({ result }) => result));
   if (!aggregate) return null;
 
-  let icon = <DiamondIcon />;
+  let icon = (
+    <DiamondIcon
+      fill={theme.base === "light" ? `${theme.color.defaultText}99` : theme.color.light}
```

Comment:
Icons use `fill="currentColor"` which means you should set the CSS `color` property to change the icon color. Color is inherited, so you can also set it an ancestor element.